### ISSUE TITLE
Product isn't displayed in order details for quantity of less than 1

### ISSUE THREAD

**Issue by reginabally** [woocommerce/woocommerce-android#2838]
**Describe the bug**
If the site has an extension installed to allow order quantity of a product in decimal, the product will not show up at the order details screen if the quantity is less than 1 (e.g. 0.3).

If the quantity is more than 1, the product shows up without the decimal quantity.

Reported in 3302211-zen.

**To Reproduce**

Pre-requisite: An extension that enabled decimal quantities should be installed and activated. 

1. Create an order with a product that has the quantity of less than 1 (e.g. 0.3)
2. Create another order with a product that has a quantity of 1.1
3. On the app, go to 'Orders'
4. Tap on the order created in (1)
5. See that the Product section is missing
6. Go back to the Orders screen and tap on the order created in (2)
7. See that the product quantity showing at the Product section is in integer (i.e. 1)

**Screenshot for Step 5**

![Screenshot_2020-09-11-14-15-09-97_10543d49da99b6f6a7dd3c9ec39fe2e4](https://user-images.githubusercontent.com/40906847/92893637-6fe21200-f44c-11ea-9069-e95477405bb2.jpg)

**Screenshot for Step 7**

![Screenshot_2020-09-11-14-15-33-56_10543d49da99b6f6a7dd3c9ec39fe2e4](https://user-images.githubusercontent.com/40906847/92893683-7bcdd400-f44c-11ea-8895-9032ad44c2c1.jpg)

**Expected behavior**
The product should be displayed on the order details screen with the correct product quantity in decimal.

**Mobile Environment**
Please include:
- Device: Oppo Reno 4
- Android version: 10
- WooCommerce Android version: 4.9 and 5.0-rc-1

**WordPress Environment**
<details>
```
`
### WordPress Environment ###

WordPress address (URL): https://woo20200908.mystagingwebsite.com
Site address (URL): https://woo20200908.mystagingwebsite.com
WC Version: 4.5.1
REST API Version: ✔ 4.5.1
WC Blocks Version: ✔ 3.1.0
Action Scheduler Version: ✔ 3.1.6
WC Admin Version: ✔ 1.5.0
Log Directory Writable: ✔
WP Version: 5.5.1
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: ✔

### Server Environment ###

Server Info: nginx
PHP Version: 7.3.22
PHP Post Max Size: 2 GB
PHP Time Limit: 1200
PHP Max Input Vars: 6144
cURL Version: 7.72.0
OpenSSL/1.1.1d

SUHOSIN Installed: –
MySQL Version: 5.5.5-10.3.24-MariaDB-log
Max Upload Size: 2 GB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 4.5.1
WC Database Prefix: wp_
Total Database Size: 2.72MB
Database Data Size: 1.20MB
Database Index Size: 1.52MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 0.14MB + Index: 0.06MB + Engine InnoDB
wp_postmeta: Data: 0.20MB + Index: 0.14MB + Engine InnoDB
wp_posts: Data: 0.06MB + Index: 0.06MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

attachment: 52
page: 6
post: 1
product: 26
product_variation: 20
shop_order: 2

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (4) ###

Akismet Anti-Spam: by Automattic – 4.1.6
Jetpack by WordPress.com: by Automattic – 8.9
All in One Product Quantity for WooCommerce: by WPWhale – 3.3 – Installed version not tested with active version of WooCommerce 4.5.1
WooCommerce: by Automattic – 4.5.1

### Inactive Plugins (0) ###


### Dropin Plugins (2) ###

advanced-cache.php: advanced-cache.php
object-cache.php: Memcached

### Settings ###

API Enabled: –
Force SSL: –
Currency: GBP (£)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –

### WC Pages ###

Shop base: #104 - /shop/
Cart: #105 - /cart/
Checkout: #106 - /checkout/
My account: #107 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Twenty Twenty
Version: 1.5
Author URL: https://wordpress.org/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Action Scheduler ###

Complete: 8
Oldest: 2020-09-08 06:26:11 +0000
Newest: 2020-09-11 06:15:30 +0000

`
```
</details>

**Comment by eduardozulian** [comment]
Also reported in the app reviews p1611911445004900-slack-C013AAPA4G0

**Comment by shiki** [comment]
This is related to https://github.com/woocommerce/woocommerce-android/issues/3611

**Comment by eduardozulian** [comment]
Another case in 3844532-zen but according to the user:

> The snippets used in my site just make decimal quantities and values lower than 1 allowed. I used woocommerce filter hooks just as it's documented here https://docs.woocommerce.com/document/adjust-the-quantity-input-values/# or here https://stackoverflow.com/questions/64861903/enable-decimal-quantities-and-stock-for-woocommerce-products

**Comment by eduardozulian** [comment]
Another case in 3889280-zen

**Comment by anitaa1990** [comment]
This was fixed in #4267 and released in v7.1
